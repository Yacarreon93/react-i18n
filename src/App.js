import React, { Component } from 'react';
import { 
  FormattedMessage,
  FormattedHTMLMessage,
  FormattedDate,
  FormattedTime,
  FormattedRelative,
  FormattedNumber,
  injectIntl,
} from 'react-intl';

import messages from './messages';
import data from './data';

class App extends Component {
  render() {
    // console.log(this.props);
    const { intl } = this.props;
    return (
      <div>
        <header>
          <h1>
           <FormattedMessage id="detail.reviewsHeading"/>
          </h1>
        </header>
        <p><FormattedMessage id="detail.author" values={{ author: <b>Yasser</b> }}/></p>
        <p><FormattedMessage id="detail.averageRating" values={{ avg: '10', count: data[0].reviews }}/></p>
        <p><FormattedMessage id="detail.purchase"/></p>
        <br />
        <br />
        <FormattedHTMLMessage id="detail.window" values={{ numMerchants: 10 }} />
        <br />
        <br />
        <br />
        <FormattedDate 
          value={new Date(1459913574887)}
          year='numeric'
          month='long'
          day='numeric'
          weekday='long'
        />
        <br />
        <br />
        <FormattedTime
          value={new Date(1459913574887)}
          year='2-digit'
          month='2-digit'
          day='2-digit'
        />
        <br />
        <FormattedRelative 
          value={new Date(1459913574887)}
          // updateInterval={1000}
          style="numeric"
        />
        <br />
        <br />
        <FormattedNumber  
          value={data[0].price[intl.locale]}
          style="currency"
          currencyDisplay="symbol"
          currency={intl.locale === 'en-US' ? 'USD' : 'EUR'}
        />
        <br />
        <br />
        <button>
          <FormattedMessage id="detail.toggle" />
        </button>
        <br />
        <br />
        <textarea placeholder={intl.formatMessage({ id: 'detail.placeholder' })}>
        </textarea>
      </div>
    );
  }
}

export default injectIntl(App);
