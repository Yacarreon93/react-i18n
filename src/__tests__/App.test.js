import React from 'react';
import { mountWithIntl, shallowWithIntl} from '../intl-enzyme';
import toJson, { renderToJson } from 'enzyme-to-json';

import App from '../App';

Date.now = jest.fn((() => 1459913574887))

let wrapper = null;

describe('app', () => {
  
  it('Render expected markup', () => {
    wrapper = mountWithIntl(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});

describe('intl messages', () => {
  beforeEach(() => {
    wrapper = shallowWithIntl(<App />).first().shallow();
  });
  
  it('Renders a single purchase message', () => {
    expect(wrapper.find({ id: 'detail.purchase' })).toHaveLength(1);
  });

  it('Passes the num of merchants to "new window" message', () => {
    expect(wrapper.find({ id: 'detail.window' }).prop('values')).toMatchObject({ numMerchants: 10 });
  });

});