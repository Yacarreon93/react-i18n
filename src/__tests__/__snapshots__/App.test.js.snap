// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app Render expected markup 1`] = `
<InjectIntl(App)
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en-US",
      "messages": Object {
        "detail.author": "by {author}",
        "detail.averageRating": "Average Rating: {avg} ({count, plural, =0 {No Reviews Yet!} one {# Review} other {# Reviews}})",
        "detail.placeholder": "What did you think?",
        "detail.purchase": "Purchase this book from:",
        "detail.reviewsHeading": "Reviews",
        "detail.toggle": "Toggle",
        "detail.window": "<small><em>All {numMerchants} links open in a new window.</em></small>",
      },
      "now": [Function],
      "textComponent": "span",
    }
  }
>
  <App
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en-US",
        "messages": Object {
          "detail.author": "by {author}",
          "detail.averageRating": "Average Rating: {avg} ({count, plural, =0 {No Reviews Yet!} one {# Review} other {# Reviews}})",
          "detail.placeholder": "What did you think?",
          "detail.purchase": "Purchase this book from:",
          "detail.reviewsHeading": "Reviews",
          "detail.toggle": "Toggle",
          "detail.window": "<small><em>All {numMerchants} links open in a new window.</em></small>",
        },
        "now": [Function],
        "textComponent": "span",
      }
    }
  >
    <div>
      <header>
        <h1>
          Â 
          <FormattedMessage
            id="detail.reviewsHeading"
            values={Object {}}
          >
            <span>
              Reviews
            </span>
          </FormattedMessage>
        </h1>
      </header>
      <p>
        <FormattedMessage
          id="detail.author"
          values={
            Object {
              "author": <b>
                Yasser
            </b>,
            }
          }
        >
          <span>
            by 
            <b>
              Yasser
            </b>
          </span>
        </FormattedMessage>
      </p>
      <p>
        <FormattedMessage
          id="detail.averageRating"
          values={
            Object {
              "avg": "10",
              "count": 1,
            }
          }
        >
          <span>
            Average Rating: 10 (1 Review)
          </span>
        </FormattedMessage>
      </p>
      <p>
        <FormattedMessage
          id="detail.purchase"
          values={Object {}}
        >
          <span>
            Purchase this book from:
          </span>
        </FormattedMessage>
      </p>
      <br />
      <br />
      <FormattedHTMLMessage
        id="detail.window"
        values={
          Object {
            "numMerchants": 10,
          }
        }
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "<small><em>All 10 links open in a new window.</em></small>",
            }
          }
        />
      </FormattedHTMLMessage>
      <br />
      <br />
      <br />
      <FormattedDate
        day="numeric"
        month="long"
        value={2016-04-06T03:32:54.887Z}
        weekday="long"
        year="numeric"
      >
        <span>
          Tuesday, April 5, 2016
        </span>
      </FormattedDate>
      <br />
      <br />
      <FormattedTime
        day="2-digit"
        month="2-digit"
        value={2016-04-06T03:32:54.887Z}
        year="2-digit"
      >
        <span>
          04/05/16, 10:32 PM
        </span>
      </FormattedTime>
      <br />
      <FormattedRelative
        style="numeric"
        updateInterval={10000}
        value={2016-04-06T03:32:54.887Z}
      >
        <span>
          2 years ago
        </span>
      </FormattedRelative>
      <br />
      <br />
      <FormattedNumber
        currency="USD"
        currencyDisplay="symbol"
        style="currency"
        value="14.95"
      >
        <span>
          $14.95
        </span>
      </FormattedNumber>
      <br />
      <br />
      <button>
        <FormattedMessage
          id="detail.toggle"
          values={Object {}}
        >
          <span>
            Toggle
          </span>
        </FormattedMessage>
      </button>
      <br />
      <br />
      <textarea
        placeholder="What did you think?"
      />
    </div>
  </App>
</InjectIntl(App)>
`;
